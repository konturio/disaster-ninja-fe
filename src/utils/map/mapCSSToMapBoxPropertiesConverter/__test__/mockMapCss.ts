export const mapCSS = {
  width: 1,
  'casing-width': 3,
  color: '#FEFEFE',
  'casing-color': 'black',
  opacity: 0.5,
  'casing-opacity': 1,
  dashes: '1,2 ,4',
  'casing-dashes': '1',
  linecap: 'round',
  'casing-linecap': 'square',
  linejoin: 'miter',
  'casing-linejoin': 'bevel',
  'fill-color': 'rgb(255, 255, 255)',
  'fill-opacity': 0.3,
  'icon-image': 'airport',
  'icon-width': 13,
  'icon-height': 12,
  'icon-opacity': 1,
  'font-family': 'Open Sans, Tahoma',
  'font-size': 16,
  'font-weight': 'bold',
  'text-color': 'gray',
  'text-opacity': 1,
  'text-position': 'center',
  'text-offset': 3,
  'max-width': 300,
  text: 'label',
  'text-halo-color': 'red',
  'text-halo-radius': 2,
  offset: 3,
  'casing-offset': 4,
};
