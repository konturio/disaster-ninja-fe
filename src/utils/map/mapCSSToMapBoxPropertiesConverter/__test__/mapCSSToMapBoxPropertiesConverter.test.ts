import { test, expect } from 'vitest';
import { mapCSSToMapBoxProperties } from '../index';
import { mapCSS } from './mockMapCss';

test('Convert mapCSS style to mapBox layers', () => {
  const result = mapCSSToMapBoxProperties(mapCSS);
  expect(result).toMatchSnapshot();
});

test('Not create extra line-offset layers', () => {
  const result = mapCSSToMapBoxProperties({
    color: 'black',
    width: 1,
    opacity: 0.2,
  });
  expect(result.length).toEqual(1);
});
