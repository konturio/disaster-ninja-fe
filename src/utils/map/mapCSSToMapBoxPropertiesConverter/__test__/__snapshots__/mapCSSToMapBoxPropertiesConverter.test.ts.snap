// Vitest Snapshot v1

exports[`Convert mapCSS style to mapBox layers 1`] = `
[
  {
    "layout": {
      "line-cap": "round",
      "line-join": "miter",
    },
    "paint": {
      "line-color": "#FEFEFE",
      "line-offset": 3,
      "line-opacity": 0.5,
      "line-pattern": [
        "1",
        "2",
        "4",
      ],
      "line-width": 1,
    },
    "type": "line",
  },
  {
    "layout": {
      "line-cap": "square",
      "line-join": "bevel",
    },
    "paint": {
      "line-color": "black",
      "line-offset": 7,
      "line-opacity": 1,
      "line-pattern": [
        "1",
      ],
      "line-width": 5,
    },
    "type": "line",
  },
  {
    "paint": {
      "fill-color": "rgb(255, 255, 255)",
      "fill-opacity": 0.3,
    },
    "type": "fill",
  },
  {
    "layout": {
      "icon-image": "airport",
      "icon-size": 12,
      "symbol-placement": "line-center",
      "text-field": "label",
      "text-font": [
        "Open Sans",
        "Tahoma",
      ],
      "text-max-width": 300,
      "text-offset": 3,
      "text-size": 16,
    },
    "paint": {
      "icon-opacity": 1,
      "text-color": "gray",
      "text-halo-color": "red",
      "text-halo-width": 2,
      "text-opacity": 1,
    },
    "type": "symbol",
  },
  {
    "paint": {
      "circle-radius": 3,
    },
    "type": "circle",
  },
]
`;
